### DISCUSSION TITLE
ilogtail有没有方法在机器负载高的时候人为触发停止log采集动作？

### DISCUSSION THREAD

**Discussion by gohonsen** [alibaba/ilogtail#559] [Help] [answered]
有这样一个场景：公司服务每月会有几天举行福利活动，在福利活动开始的前几分钟会瞬间涌入大量流量，将服务带宽占满，机器负载飙升，我们想在这几分钟高峰时间内让日志采集工具停止采集动作，等高峰时间过去之后，恢复采集；目前看 ilogtail 的资料，社区版不支持中心化配置管理，系统参数里面的 cpu_usage_limit/mem_usage_limit  也都是软限制，缓存也会带来 IO 负载，有没有什么好的办法可以主动触发 ilogtail 停止/恢复采集 ？

**Comment by liuhaoyang** [comment]
目前可以通过 `LOGTAIL_HTTP_LOAD_CONFIG` 环境变量来开启 HTTP 配置更新的接口，通过 API 调用去暂停收集。
相关的接口参考源码  https://github.com/alibaba/ilogtail/blob/main/plugin_main/plugin_http.go

**Reply by yyuuttaaoo** [reply]
这个功能是给测试用的，可能不太完善，有以下几个问题
1. 没有对应的resume API
2. holdon调用是同步的，可能返回时间比较长
3. 配置热加载会holdon，如果此时正好请求holdon API会导致插件崩溃
4. 无法控制C++采集的行为
应该还不能用到生产环境，需要升级一下。

**Reply by gohonsen** [reply]
你好，从你发的链接看，只看到对CPU 和内存的一些 dump 操作接口，没有找到控制采集行为的 API

**Comment by XLPE** [comment, marked as answer]
简单描述一下我的思路： 我这里做了一些定制（也可以等社区版的远程配置加载），远程停止采集配置的下发(模拟暂停）就行了。不过这个做法可能会导致checkpoint丢失，还需要修改一下checkpoint的存储机制，做到根据超时时间来删除。
